libs/model/src/aggregates/user/signIn/utils.ts

```diff
@@ -0,0 +1,328 @@
+  ssoInfo: VerifiedUserInfo,
+) {
+  let query = `
+    WHERE user_id IS NOT NULL
+      AND oauth_provider = :oauthProvider
+      AND
+  `;
+
+  if (['github', 'discord', 'twitter'].includes(ssoInfo.provider)) {
+    query += ` oauth_username = :oauthUsername`;
+  } else if (['google', 'email', 'apple'].includes(ssoInfo.provider)) {
+    query += `oauth_email = :oauthEmail`;
+  } else if (['phone_number'].includes(ssoInfo.provider)) {
+    query += `oauth_phone_number = :oauthPhoneNumber`;
+  } else {
+    throw new Error(`Unsupported OAuth provider: ${ssoInfo.provider}`);
+  }
+
+  return query;
+}
+
+export async function findUserByAddressOrHex(
+  searchTerm: { address: string } | { hex: string },
+  transaction: Transaction,
+): Promise<UserAttributes | null> {
+  log.trace(`findUserByAddressOrHex: ${JSON.stringify(searchTerm)}`);
+  const users = await models.sequelize.query(
+    `
+      WITH user_ids AS (SELECT DISTINCT(user_id) as user_id
+                        FROM "Addresses"
+                        WHERE user_id IS NOT NULL
+                          AND ${'address' in searchTerm ? 'address=:address' : 'hex=:hex'})
+      SELECT U.*
+      FROM "Users" U
+      WHERE U.id IN (SELECT user_id FROM user_ids)
+    `,
+    {
+      model: models.User,
+      replacements: searchTerm,
+      transaction,
+    },
+  );
+
+  if (users.length > 1)
+    throw new Error(
+      `Multiple users with the same ${Object.keys(searchTerm)[0]} found!`,
+    );
+  if (users.length === 0) return null;
+  return users[0];
+}
+
+export async function findUserBySso(
+  ssoInfo: VerifiedUserInfo,
+  transaction: Transaction,
+): Promise<UserAttributes | null> {
+  log.trace(`findAddressesBySso: ${JSON.stringify(ssoInfo)}`);
+  const users = await models.sequelize.query(
+    `
+      WITH user_ids AS (SELECT DISTINCT(user_id) as user_id
+                        FROM "Addresses" ${constructFindAddressBySsoQueryFilter(ssoInfo)})
+      SELECT U.*
+      FROM "Users" U
+      WHERE U.id IN (SELECT user_id FROM user_ids);
+    `,
+    {
+      model: models.User,
+      replacements: {
+        oauthProvider: ssoInfo.provider,
+        oauthUsername: ssoInfo.username,
+        oauthEmail: ssoInfo.email,
+        oauthPhoneNumber: ssoInfo.phoneNumber,
+      },
+      transaction,
+    },
+  );
+
+  if (users.length > 1)
+    throw new Error('Multiple users with the same oauth info found!');
+  if (users.length === 0) return null;
+  return users[0];
+}
+
+export async function findOrCreateUser({
+  address,
+  transaction,
+  ssoInfo,
+  referrer_address,
+  privyUserId,
+  hex,
+  signedInUser,
+}: {
+  address: string;
+  transaction: Transaction;
+  ssoInfo?: VerifiedUserInfo;
+  referrer_address?: string | null;
+  privyUserId?: string;
+  hex?: string;
+  signedInUser?: UserAttributes | null;
+}): Promise<{
+  newUser: boolean;
+  user: UserAttributes;
+}> {
+  let foundUser: UserAttributes | null;
```

Comment:
you don't like ternary :-|

Reply:
I'm usually not a fan of ternary unless the options are short. If prettier wraps the function calls to multiple lines I find ternary to be less readable than a traditional `if` conditional. In this case prettier wraps it nicely so I will update.